src/3rdparty/walkontable/src/renderer/rows.js

```diff
@@ -54,6 +55,16 @@ export default class RowsRenderer extends BaseRenderer {
         of rendered rows by specifying the table height and/or turning off the "renderAllRows" option.`);
     }
 
+    const thead = this.rootNode.previousElementSibling;
+    if (thead) {
+      const isTheadEmpty = thead.firstElementChild === null;
+      if (isTheadEmpty) {
+        addClass(this.rootNode, 'afterEmptyThead');
+      } else {
+        removeClass(this.rootNode, 'afterEmptyThead');
+      }
+    }
+
```

Comment:
The "rows" renderer is responsible for adjusting the DOM nodes only for TR elements within TBODY. Traversing within this module/renderer to different DOM nodes should be avoided. I've prepared a commit where I moved that logic to the TableRenderer. It seems that the renderer has all the needed information about the state so no DOM checking is necessary to toggle the CSS class.

All tests pass https://github.com/handsontable/handsontable/commit/1d2b5d8d57bc5f68a825b044df9f7ef4ebd48235 (except lint)